fix #8: move fixup function to the global ns

seletskiy opened the pull request:
Make possible to remap <C-O> key in insert mode without breaking
functionality of this plugin.

Like this:

``` viml
au VimEnter * inoremap <expr> <C-O> (pumvisible()
    \ ? feedkeys("\<C-N>")
    \ : feedkeys("\<C-R>=g:MatchemRepeatFixupFlush('<C-O>')\<CR>\<C-O>", 'n')) ? '' : ''
```

seletskiy commented:
Actually, there are a lot of other mappings in the plugin, which also interferring with user mappings, like `(`.

So, I'll need to improve the PR.

ervandew commented:
> Actually, there are a lot of other mappings in the plugin, which also interferring with user mappings, like `(`.

So what are you doing with a `(` mapping? If your mapping the starting characters already, then I'd imagine it would make using this and other similar plugins very difficult.

kovetskiy commented:
@ervandew I'm using something like `inoremap ( <C-O>:call MyAutocompleteHelper()<CR>`

ervandew commented:
In that case perhaps a buffer local setting should be added to matchem to define the matching chars to map (defaulting to `&matchpairs`).  That way your autocomplete helper can be responsible for the closing paren instead of trying to get matchem to work with it (and any number of other permutations other users might have mapped).

seletskiy commented:
@ervandew: `MyAutocompleteHelper()` can be responsible for enclosing paren, but if in some circumstances it's need to fallback for default matchem behaviour, there should be a way of calling required matchem functions, that are declared private just now.

seletskiy committed:
Fix matchem integration with UltiSnips.

Also, export all fixup functions to public namespace to make possible of
remapping all keys which are hold by matchem without breaking it's
functionality.

seletskiy commented:
@ervandew: ping! I've added UltiSnips integration (matchem did not work correctly), also, I've moved all mapped functions to public namespace to allow user for mapping matchem along with other stuff.

seletskiy commented:
@ervandew: UltiSnips issue addressed in #10 first comment with git.

seletskiy commented:
Oops, sorry, do not accept. Found some minor bug.

seletskiy committed:
restore original s-mode mappings

seletskiy commented:
@ervandew: ping?

kovetskiy commented:
@ervandew ping
